### PR TITLE
Remove orders of a disconnected peer

### PR DESCRIPTION
**Pull request by itsbalamurali**
Implements #168.

### CODE REVIEW

File: lib/orderbook/MatchingEngine.ts [hunk 1]
```diff
@@ -147,10 +147,25 @@ class MatchingEngine {
     ).add(order);
   }
 
+  public dropPeerOrders = (hostId: number): void => {
```
[the remark this answers is not in the document]
Comment (itsbalamurali) [reply lib/orderbook/MatchingEngine.ts]: done

**Comment by itsbalamurali** [comment]
@moshababo done with pr changes, please have a look

File: test/unit/MatchingEngine.spec.ts [hunk 2]
```diff
@@ -179,3 +179,20 @@ describe('MatchingEngine.match', () => {
...
   });
 });
+
+describe('MatchingEngine.dropPeerOrders', () => {
+  it('should drop the peer order from queue', () => {
+    const engine = new MatchingEngine(PAIR_ID);
+    const peerOrder = createOrder(5, -5);
+    const ownOrder = createOrder(5, 15);
+    const hostId = 1;
+    engine.addPeerOrder(peerOrder);
```
Comment (sangaman) [inline test/unit/MatchingEngine.spec.ts]: What happens if you add multiple orders for the same peer?

File: lib/orderbook/MatchingEngine.ts [hunk 3]
```diff
@@ -129,6 +130,25 @@ class MatchingEngine {
...
+    this.priorityQueues.buyOrders.remove(hostId, MatchingEngine.hostIdComparator);
+    this.priorityQueues.sellOrders.remove(hostId, MatchingEngine.hostIdComparator);
+  }
+
+  private static hostIdComparator = (queuedOrder: any, orderHostId: any) => {
```
[the remark this answers is not in the document]
Comment (itsbalamurali) [reply lib/orderbook/MatchingEngine.ts]: one of them is number which is hostId but where as in their comparator they are evaluating by alternating the parameters passed to it.

**Comment by moshababo** [comment]
@itsbalamurali let's wait with this one until `removeMany` will be implemented in our `fastpriorityqueue` [fork](https://github.com/ExchangeUnion/FastPriorityQueue.js). 

Will go through the other stuff soon.

**Comment by moshababo** [comment]
I tested the p2p context manually, and it seems to be working fine. 
The only issue is that this depends on attaching the host object to the peer, which is at the moment not a stable code, and is subject to change. I added a warning log, so we won't overlook that problem.

**Comment by moshababo** [comment]
I tested the p2p context manually, and it seems to be working fine.
The only issue is that this depends on attaching the host object to the peer, which is at the moment not a stable code, and is subject to change. I added a warning log, so we won't overlook that problem.

**Pull request merged**